src/Microsoft.Restier.Core/Conventions/ConventionBasedChangeSetItemFilter.cs

```diff
@@ -86,51 +86,55 @@ private Task InvokeProcessorMethodAsync(SubmitContext context, ChangeSetItem ite
             var expectedMethod = targetApiType.GetQualifiedMethod(expectedMethodName);
-            if (!IsUsable(expectedMethod))
+
+            if (expectedMethod == null)
             {
-                if (expectedMethod != null)
-                {
-                    Trace.WriteLine($"Restier Filter found '{expectedMethodName}' but it is unaccessible due to its protection level. Your method will not be called until you change it to 'protected internal'.");
-                }
-                else
+                var actualMethodName = expectedMethodName.Replace(dataModification.ExpectedResourceType.Name, dataModification.ResourceSetName);
+                var actualMethod = targetApiType.GetQualifiedMethod(actualMethodName);
+                if (actualMethod != null)
                 {
-                    var actualMethodName = expectedMethodName.Replace(dataModification.ExpectedResourceType.Name, dataModification.ResourceSetName);
-                    var actualMethod = targetApiType.GetQualifiedMethod(actualMethodName);
-                    if (actualMethod != null)
-                    {
-                        Trace.WriteLine($"BREAKING: Restier Filter expected'{expectedMethodName}' but found '{actualMethodName}'. Your method will not be called until you correct the method name.");
-                    }
+                    Trace.WriteLine($"Restier Filter expected'{expectedMethodName}' but found '{actualMethodName}'. Your method will not be called until you correct the method name.");
                 }
+
+                return Task.CompletedTask;
+            }
+
+            if (!expectedMethod.IsFamily && !expectedMethod.IsFamilyOrAssembly)
+            {
+                Trace.WriteLine($"Restier Filter found '{expectedMethod}' but it is inaccessible due to its protection level. Your method will not be called until you change it to 'protected internal'.");
+                return Task.CompletedTask;
+            }
+
+            if (expectedMethod.ReturnType != typeof(void) && !typeof(Task).IsAssignableFrom(expectedMethod.ReturnType))
+            {
+                Trace.WriteLine($"Restier Filter found '{expectedMethod}' but it does not return void or a Task. Your method will not be called until you correct the return type.");
+                return Task.CompletedTask;
             }
-            else
+
+            object target = null;
+            if (!expectedMethod.IsStatic)
             {
-                object target = null;
-                if (!expectedMethod.IsStatic)
+                target = context.Api;
+                if (target == null || !targetApiType.IsInstanceOfType(target))
                 {
-                    target = context.Api;
-                    if (target == null || !targetApiType.IsInstanceOfType(target))
-                    {
-                        return Task.WhenAll();
-                    }
+                    Trace.WriteLine("The Restier API is of the incorrect type.");
```

Comment:
That is a good question. I've not compared these changes against the new DI implementation. I do see the extension methods that we provide to register everything as "optional" in the sense that if someone wants to register everything themselves, they should be able to do that. An actual check makes sense.
Please note that I only changed this code a bit because I now ensure that the context is not null, and I know API won't be null on the context, so we don't have to check it anymore. The presentation of the diff on github is slightly odd.